tests/unit/lms/views/admin/lti_registration_test.py

```diff
@@ -0,0 +1,156 @@
+from unittest.mock import sentinel
+
+import pytest
+from sqlalchemy.exc import IntegrityError
+
+from lms.views.admin.lti_registration import AdminLTIRegistrationViews
+from tests import factories
+from tests.matchers import Any, temporary_redirect_to
+
+
+@pytest.mark.usefixtures(
+    "pyramid_config", "application_instance_service", "lti_registration_service"
+)
+class TestAdminApplicationInstanceViews:
+    def test_registrations(self, views):
+        assert views.registrations() == {}
+
+    def test_registration_new(self, views):
+        assert views.registration_new() == {}
+
+    def test_registration_new_post(
+        self, pyramid_request_new_registration, lti_registration_service
+    ):
+        response = AdminLTIRegistrationViews(
+            pyramid_request_new_registration
+        ).registration_new_post()
+
+        lti_registration_service.create_registration.assert_called_once_with(
+            issuer="ISSUER",
+            client_id="CLIENT_ID",
+            auth_login_url="AUTH_LOGIN_URL",
+            key_set_url="KEY_SET_URL",
+            token_url="TOKEN_URL",
+        )
+
+        assert response == temporary_redirect_to(
+            pyramid_request_new_registration.route_url(
+                "admin.registration.id",
+                id_=lti_registration_service.create_registration.return_value.id,
+            )
+        )
+
+    def test_registration_new_post_duplicate(
+        self, pyramid_request_new_registration, lti_registration_service
+    ):
+        lti_registration_service.create_registration.side_effect = IntegrityError(
+            Any(), Any(), Any()
+        )
+
+        response = AdminLTIRegistrationViews(
+            pyramid_request_new_registration
+        ).registration_new_post()
+
+        assert response.status_code == 400
+
+    @pytest.mark.parametrize(
+        "missing", ["issuer", "client_id", "auth_login_url", "key_set_url", "token_url"]
+    )
+    def test_registration_new_post_missing_params(
+        self, pyramid_request_new_registration, missing
+    ):
+        del pyramid_request_new_registration.params[missing]
+
+        response = AdminLTIRegistrationViews(
+            pyramid_request_new_registration
+        ).registration_new_post()
+
+        assert response.status_code == 400
+
+    def test_search_not_query(self, pyramid_request):
+        response = AdminLTIRegistrationViews(pyramid_request).search()
```

Comment:
```suggestion
    def test_search_not_query(self, views, pyramid_request):
        response = views.search()
```

The same seems true for all the rest of the views which don't use the new stuff either.